OSDOCS-10271: VMware CPMS TP to GA (#75106)
openshift/openshift-docs: modules/cpmso-yaml-failure-domain-vsphere.adoc

:_mod-docs-content-type: REFERENCE
[id="cpmso-yaml-failure-domain-vsphere_{context}"]
= Sample {vmw-full} failure domain configuration

On {vmw-full} infrastructure, the cluster-wide infrastructure Custom Resource Definition (CRD), `infrastructures.config.openshift.io`, defines failure domains for your cluster.

> The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains.

Reviewer: ```suggestion
The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains which the control plane machineset will use to ensure control plane nodes are deployed to the appropriate failure domain.
```

A failure domain is an infrastructure resource made up of a control plane machine set, a vCenter datacenter, vCenter datastore, and a network.

> By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on hardware that is separate from the primary {vmw-full} infrastructure.

Reviewer: ```suggestion
By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on  separate clusters or datacenters.  
```

[IMPORTANT]
====
Each `failureDomains.platform.vsphere.name` field value in the `ControlPlaneMachineSet` CR must match the corresponding value defined in the `failureDomains.name` field of the cluster-wide infrastructure CRD.
Currently, the `vsphere.name` field is the only supported failure domain field that you can specify in the `ControlPlaneMachineSet` CR.

> ====

Reviewer 2: LGTM , if it helps we can add below  to get failure domain name from infrastructure CRD.
`oc get infrastructure cluster -o=jsonpath={.spec.platformSpec.vsphere.failureDomains[0].name}`

Author: That's a great thing to add, thanks!
